vendor/vendor_init.te

```diff
@@ -2,3 +2,4 @@ allow vendor_init self:capability2 block_suspend;
 allow vendor_init kernel:system module_request;
 allow vendor_init debugfs_tracing_instances:dir create_dir_perms;
 allow vendor_init debugfs_tracing_instances:file w_file_perms;
+allow vendor_init proc:file write;
```

Comment:
we can declare a new file label for swappiness and disk_based_swap:
https://cactus-source.iind.intel.com/source/xref/CELADON-MASTER-R2/device/google/redbull-sepolicy/vendor/google/genfs_contexts#12
then add the following rule:
https://cactus-source.iind.intel.com/source/xref/CELADON-MASTER-R2/device/google/redbull-sepolicy/vendor/google/vendor_init.te#6

Reply:
Hi @Shao-Feng
Can u pls let us know what is the  sys entry for swappiness  for our Platform.
I have tried with /sys/vm/swappiness but i cant see this entry in device through adb shell
and device test cases are failing.

Reply:
/proc/sys/vm/swappiness is this the proc file name which i can use in genfs_contexts

genfscon proc /proc/sys/vm/swappiness		 u:object_r:proc_swappiness:s0 like this?

Reply:
in file.te add:
type proc_swappiness, proc_type, fs_type;
in genfs_contexts add:
genfscon proc /sys/vm/swappiness                      u:object_r:proc_swappiness:s0
in vendor_inite.te add:
allow vendor_init proc_swappiness:file w_file_perms;

Reply:
tried exactly the same way what you told.
Still the test cases are failing.

Reply:
do you add same rules for disk_based_swap?

Reply:
no I have just added what mentioned for swappiness.
is different proc entry for disk_based_swap?
If I am not right you are asking to add the the same thing for

/proc/sys/vm/disk_based_swap?

Reply:
I only noticed these is an AVC error in the committed message:
avc: denied { write } for comm="init" name="disk_based_swap" dev="proc"
ino=22698 scontext=u:r:vendor_init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=0

you should to figure out where is the disk_based_swap.

Reply:
08-10 07:28:46.449   192   192 I auditd  : type=1400 audit(0.0:14): avc: denied { write } for comm="init" name="swappiness" dev="proc" ino=22697 scontext=u:r:vendor_init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=0

08-10 07:28:46.449   192   192 I auditd  : type=1400 audit(0.0:15): avc: denied { write } for comm="init" name="disk_based_swap" dev="proc" ino=22698 scontext=u:r:vendor_init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=0

08-10 07:28:46.453   392   392 I boot_progress_start: 7669

we have two denials and we kept only one otherwise commit message will be go longer.
So we need to add same way for disk_based_swap?

Reply:
yes, and please figure out the right path in proc for disk_based_swap.